### PR TITLE
Added else if

### PR DESCRIPTION
**Pull request by Preciseful**

### CODE REVIEW

File: app/javascript/src/lib/OWLanguageLinter.js [hunk 1]
```diff
@@ -496,17 +496,17 @@ function findConditionalsRegexErrors(content) {
...
 
   let match
   while ((match = regex.exec(content)) != null) {
-    const usageFormatted = match[0].replace(/[\s\n]+/, " ")
     diagnostics.push({
       from: match.index,
       to: match.index + match[0].length,
       severity: "error",
-      message: `\`${ usageFormatted }\` is not supported. Use \`@else { @if (elseIfCondition) { ... } }\` instead.`
+      // maybe update the error message? i'm just not sure what to
```
Comment (netux) [inline app/javascript/src/lib/OWLanguageLinter.js:508]: I think it's fine.

**Review by netux — APPROVED** [review]
LGTM!

File: app/javascript/src/lib/OWLanguageLinter.js [hunk 2]
```diff
@@ -496,17 +496,17 @@ function findConditionalsRegexErrors(content) {
...
+      message: "@elif is not valid. Use `@else if (elseIfCondition) { ... }` instead."
     })
```
Comment (Mitcheljager) [inline app/javascript/src/lib/OWLanguageLinter.js:510]: I think we can do without this whole error, we don't need an error for something that doesn't exist

File: app/javascript/src/lib/OWLanguageLinter.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ export function OWLanguageLinter(view) {
   findMissingComparisonsInConditions(content)
   findTrailingCommas(content)
   findConditionalsRegexErrors(content)
-  findConditionalsElseIfUses(content)
+  findConditionalsElifUses(content)
```
Comment (Mitcheljager) [inline app/javascript/src/lib/OWLanguageLinter.js:20]: Let's use the full name for this function

File: app/javascript/src/lib/OWLanguageLinter.js [hunk 4]
```diff
@@ -496,17 +496,17 @@ function findConditionalsRegexErrors(content) {
   }
 }
 
-function findConditionalsElseIfUses(content) {
-  const regex = /@else[\s\n]+@?if/g
+function findConditionalsElifUses(content) {
```
Comment (Mitcheljager) [inline app/javascript/src/lib/OWLanguageLinter.js:499]: Let's use the full name for this function

File: app/javascript/src/utils/compiler.js [hunk 5]
```diff
@@ -282,15 +283,54 @@ function evaluateConditionals(joinedItems) {
...
+    let trueBlockContent = joinedItems.substring(openingBracketIndex + 1, closingBracketIndex)
     let falseBlockContent = ""
+    let passed = evaluateExpressionTree(conditionExpressionTree)
+
+    elifStartRegex.lastIndex = closingBracketIndex
+    let elifMatch = elifStartRegex.exec(joinedItems)
+
+    while (elifMatch != null && elifMatch.index === closingBracketIndex + 1) {
+      const openingElifConditionParenIndex = elifMatch.index + elifMatch[0].length - 1
+      const closingElifConditionParenIndex = getClosingBracket(joinedItems, "(", ")", openingElifConditionParenIndex - 1)
```
Comment (Mitcheljager) [inline app/javascript/src/utils/compiler.js:302]: Lets use the full name instead of `Elif`

File: app/javascript/src/utils/compiler.js [hunk 6]
```diff
@@ -282,15 +283,54 @@ function evaluateConditionals(joinedItems) {
...
+
+      const openingElifBracketIndex = startElifBracketMatch.index + startElifBracketMatch[0].length - 1
+      closingBracketIndex = getClosingBracket(joinedItems, "{", "}", openingElifBracketIndex - 1)
+      if (closingBracketIndex < 0) {
+        break
+      }
+
+      conditionalEndingIndex = closingBracketIndex
+
+      if(!passed) {
```
Comment (Mitcheljager) [inline app/javascript/src/utils/compiler.js:324]: Missing space after if 
```suggestion
      if (!passed) {
```

File: app/javascript/src/utils/compiler.js [hunk 7]
```diff
@@ -282,15 +283,54 @@ function evaluateConditionals(joinedItems) {
...
+        passed = evaluateExpressionTree(conditionElifExpressionTree)
+        if(passed) {
```
Comment (Mitcheljager) [inline app/javascript/src/utils/compiler.js:326]: Missing space after if 
```suggestion
        if (passed) {
```

**Review by Mitcheljager — CHANGES_REQUESTED** [review]
Nice! Left a few formatting suggestions, but looks good overall!

File: app/javascript/src/lib/OWLanguageLinter.js (hunk 4 above)
Comment (netux) [reply app/javascript/src/lib/OWLanguageLinter.js:499]: Psst: I think this is on purpose. It is finding literally "elif", not "else if".

Comment (Mitcheljager) [reply app/javascript/src/lib/OWLanguageLinter.js:499]: Oh of course this is the linter, I got it confused with the compiler

File: app/javascript/src/lib/OWLanguageLinter.js (hunk 2 above)
Comment (Preciseful) [reply app/javascript/src/lib/OWLanguageLinter.js]: okay, i removed it

File: app/javascript/src/utils/compiler.js [hunk 8]
```diff
@@ -263,6 +263,7 @@ function evaluateConditionals(joinedItems) {
   const ifStartRegex = /@if[\s\n]*\(/g
   const startBracketRegex = /[\s\n]*\{/g
   const elseStartRegex = /[\s\n]*@else[\s\n]*\{/g
+  const elseifStartRegex = /[\s\n]*@else if[\s\n]*\(/g
```
Comment (netux) [inline app/javascript/src/utils/compiler.js:266]: `elseIf`*, capital `I` in "If"

**Review by Mitcheljager — APPROVED** [review]

**Pull request merged**